code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -46,4 +44,6 @@ const StageProgress = React.createClass({
     );
   }
 });
-module.exports = StageProgress;
+export default connect((state, ownProps) => ({
+  levels: ownProps.levels || state.stages[0].levels
```

Comment:
Oh, this is already supported.  As long as you have the `Provider` component wrapping your rendering root, you can connect at any layer with no concern for whether parents or children are connected.  Basically, `Provider` makes your store a global to all of its descendants, and `connect()` consumes that now-"global" store.  It uses the React [context](https://facebook.github.io/react/docs/context.html) feature - see Redux videos [#25](https://egghead.io/lessons/javascript-redux-passing-the-store-down-implicitly-via-context) and [#26](https://egghead.io/lessons/javascript-redux-passing-the-store-down-with-provider-from-react-redux) for details.